src/main/java/de/tum/cit/aet/artemis/modeling/dto/UpdateModelingExerciseDTO.java

```diff
@@ -0,0 +1,133 @@
+package de.tum.cit.aet.artemis.modeling.dto;
+
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import jakarta.validation.constraints.Positive;
+
+import org.hibernate.Hibernate;
+import org.jspecify.annotations.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.cit.aet.artemis.assessment.domain.GradingCriterion;
+import de.tum.cit.aet.artemis.assessment.dto.GradingCriterionDTO;
+import de.tum.cit.aet.artemis.exercise.domain.DifficultyLevel;
+import de.tum.cit.aet.artemis.exercise.domain.IncludedInOverallScore;
+import de.tum.cit.aet.artemis.modeling.domain.ModelingExercise;
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record UpdateModelingExerciseDTO(long id, @Nullable String title, @Nullable String channelName, @Nullable String shortName, @Nullable String problemStatement,
+        @Nullable Set<String> categories, @Nullable DifficultyLevel difficulty, @Nullable @Positive Double maxPoints, @Nullable @Positive Double bonusPoints,
+        @Nullable IncludedInOverallScore includedInOverallScore, @Nullable ZonedDateTime releaseDate, @Nullable ZonedDateTime startDate, @Nullable ZonedDateTime dueDate,
+        @Nullable ZonedDateTime assessmentDueDate, @Nullable ZonedDateTime exampleSolutionPublicationDate, @Nullable String exampleSolutionModel,
+        @Nullable String exampleSolutionExplanation, @Nullable Long courseId, @Nullable Long exerciseGroupId, @Nullable Set<GradingCriterionDTO> gradingCriteria,
+        @Nullable String gradingInstructions) {
+
+    /**
+     * Applies this DTO's fields to the given exercise, mutating it in place.
+     *
+     * @param exercise the exercise to update (will be mutated)
+     * @return the same exercise instance after applying updates
+     */
+    public ModelingExercise update(ModelingExercise exercise) {
+        if (this.title != null) {
+            exercise.setTitle(this.title);
+        }
+        if (this.channelName != null) {
+            exercise.setChannelName(this.channelName);
+        }
+        if (this.shortName != null) {
+            exercise.setShortName(this.shortName);
+        }
+        if (this.problemStatement != null) {
+            exercise.setProblemStatement(this.problemStatement);
+        }
+        if (this.categories != null) {
+            exercise.setCategories(this.categories);
+        }
+        if (this.difficulty != null) {
+            exercise.setDifficulty(this.difficulty);
+        }
+        if (this.maxPoints != null) {
+            exercise.setMaxPoints(this.maxPoints);
+        }
+        if (this.bonusPoints != null) {
+            exercise.setBonusPoints(this.bonusPoints);
+        }
+        if (this.includedInOverallScore != null) {
+            exercise.setIncludedInOverallScore(this.includedInOverallScore);
+        }
+        if (this.releaseDate != null) {
```

Comment:
this prevents that end users can set the release date to null, which should be possible